packages/backend/migrations/20230918113102_rename_payment_pointer_tables.js

```diff
@@ -24,7 +24,15 @@ exports.up = function (knex) {
       table.renameColumn('paymentPointerId', 'walletAddressId')
       table.foreign('walletAddressId').references('walletAddresses.id')
       table.index(['walletAddressId', 'createdAt', 'id'])
-    })
+    }),
+    knex('webhookEvents')
+      .update({
+        // renames paymentPointer keys (if any) to walletAddress in data json column
+        data: knex.raw(
+          "data - 'paymentPointer' || json_build_object('walletAddress', data->'paymentPointer')"
+        )
+      })
+      .whereRaw("data \\? 'paymentPonter'")
```

Comment:
I'm getting an error when this migration runs: `operator does not exist: json ? unknown` because this column is `json` (not `jsonb`) which doesnt have the `?` operator. Similar errors are lurking behind this one. This is working for me, although perhaps not the most efficient because its casting the `json` to `jsonb` twice:

```js
    knex('webhookEvents')
      .update({
        data: knex.raw(
          "data::jsonb -'paymentPointer' || jsonb_build_object('walletAddress', data::jsonb->'paymentPointer')"
        )
      })
      .whereRaw("data->'paymentPointer' is not null")
```
This is working for me as well and should be better because it casts once in a CTE and then references that in the update. Harder to read though. 🤷 

```js
    knex.raw(`
      WITH cte_data AS (
        SELECT
          id,
          data::jsonb AS json_data
        FROM
          "webhookEvents"
        WHERE
          data->'paymentPointer' IS NOT NULL
      )
      UPDATE "webhookEvents" w
      SET
        "data" = (cte_data.json_data - 'paymentPointer' || jsonb_build_object('walletAddress', cte_data.json_data->'paymentPointer'))
      FROM cte_data
      WHERE
        w.id = cte_data.id;
    `)
```
Not sure how much we should care about the performance difference with testnet as the only deployment, but casting to jsonb is relatively expensive compared to the other operations we are doing here.

Reply:
Perhaps we can handwave the performance differences since this would only run once per deployment. I'll see if there are other approaches that may work as well.